<?xml version="1.0"?>
<launch>
    <!-- 定义一个参数 group_name -->
    
    
    <arg name="group_name" default="elfin_arm"/>
    <param name="enable_trajectory_save" value="true" type="bool" />
    <param name="joint_speed_factor" value="1.0" type="double" />

    <node pkg="moveit_terminal"
          type="arm_terminal_node"
          name="arm_terminal_node"
          output="screen">
        
        <param name="group_name" value="$(arg group_name)" type="string" />
        <param name="reference_link" value="default" type="string" />

        <!-- 笛卡尔参数 -->
        <param name="eef_step" value="0.01" type="double" />
        <param name="jump_threshold" value="0" type="double" />
        <param name="try_times" value="10" type="int" />
        <param name="trajectory_try_times" value="30" type="int" />
        <param name="speed_factor" value="1" type="double" />
        <param name="singularity_threshold" value="0.01" type="double" />
        <param name="velocity_diff_torlerance" value="0.15" type="double" />
            
        <!-- 选择 move line method -->
        <!-- move_line -->
        <!-- move_line_middle_interpolation -->
        <!-- move_line_joint_complete -->
        <!-- move_line_end_disturbance -->
        <param name="move_line_method" value="move_line_middle_interpolation" type="string" />    

    </node>
    
    <node pkg="moveit_terminal"
          type="trajectory_data_visualization.py"
          name="trajectory_data_visualization"
          output="screen">

          <param name="folder_path" value="/root/work_place/moveit_utils_ws/src/moveit_utils/data" type="string" />
    </node>


</launch>